src/theory/uf/cardinality_extension.cpp

```diff
@@ -873,8 +875,8 @@ void SortModel::assertCardinality(int c, bool val)
         }
       }
       // we assert it positively, if its beyond the bound, abort
-      if (options::ufssAbortCardinality() != -1
-          && c >= options::ufssAbortCardinality())
+      if (options::ufssAbortCardinality() >= 0
+          && static_cast<int>(c) >= options::ufssAbortCardinality())
```

Comment:
This is ugly. This is better since cardinality can't be negative anyways:
```suggestion
          && c >= static_cast<uint32_t>(options::ufssAbortCardinality()))
```